src/components/location/location.service.ts

```diff
@@ -71,30 +62,26 @@ export class LocationService {
     });
 
     const result = await this.repo.readOne(id);
-    return await this.secure(result, session);
+    return this.secure(result, session);
   }
 
   async readMany(ids: readonly ID[], session: Session) {
     const locations = await this.repo.readMany(ids);
-    return await Promise.all(locations.map((dto) => this.secure(dto, session)));
+    return locations.map((dto) => this.secure(dto, session));
   }
 
-  private async secure(
-    dto: UnsecuredDto<Location>,
-    session: Session,
-  ): Promise<Location> {
+  private secure(dto: UnsecuredDto<Location>, session: Session) {
     return this.privileges.for(session, Location).secure(dto);
   }
 
   async update(input: UpdateLocation, session: Session): Promise<Location> {
-    const location = await this.readOne(input.id, session);
+    const location = await this.repo.readOne(input.id);
 
     const changes = this.repo.getActualChanges(location, input);
     this.privileges.for(session, Location, location).verifyChanges(changes);
```

Comment:
@CarsonF - Do have any insight into what the two errors here are about?

Reply:
Fixed here #3076